IdentityCore/src/MSIDError.m

```diff
@@ -99,10 +99,30 @@ MSIDErrorCode MSIDErrorCodeForOAuthError(NSString *oauthError, MSIDErrorCode def
     {
         return MSIDErrorServerAccessDenied;
     }
-    
+    if (oauthError && [oauthError caseInsensitiveCompare:@"tokenTransferFailedOTC"] == NSOrderedSame)
+    {   // Account Transfer session time out, When the token's time is expired
+        return MSIDErrorUserCancel;
+    }
     return defaultCode;
 }
 
+MSIDErrorCode MSIDErrorCodeForOAuthErrorWithSubErrorCode(NSString *oauthError, MSIDErrorCode defaultCode, NSString *subError)
```

Comment:
This looks like a generic function, but it doesn't address a generic suberror case. Should it be called something different to indicate that's specific for account transfer only?

Reply:
We created this function to have certain reply to a certain kind of error and sub error combination, If they don't occur then fall back to original implementation, Thats why we called it everywhere its called as the original method don't use the subError it should not effect original implementaiton,